core/src/appleMain/kotlin/CentralManager.kt

```diff
@@ -41,6 +41,12 @@ public class CentralManager internal constructor() {
         if (cbCentralManager.isScanning) cbCentralManager.stopScan()
     }
 
+    internal fun retrievePeripheral(withIdentifier: Uuid): CBPeripheral? {
+        val peripherals = cbCentralManager.retrievePeripheralsWithIdentifiers(listOf(withIdentifier.toNSUUID()))
+        if (peripherals.isEmpty()) return null
+        return peripherals.first() as CBPeripheral
+    }
```

Comment:
We can simplify this slight as:

```suggestion
    internal fun retrievePeripheral(withIdentifier: Uuid): CBPeripheral? =
        cbCentralManager
            .retrievePeripheralsWithIdentifiers(listOf(withIdentifier.toNSUUID()))
            .firstOrNull() as? CBPeripheral
```